src/libraries/Common/src/System/TimeProvider.cs

```diff
@@ -0,0 +1,207 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    /// <summary>Provides an abstraction for time.</summary>
+    public abstract class TimeProvider
+    {
+        private readonly double _timeToTicksRatio;
+
+        /// <summary>
+        /// Gets a <see cref="TimeProvider"/> that provides a clock based on <see cref="DateTimeOffset.UtcNow"/>,
+        /// a time zone based on <see cref="TimeZoneInfo.Local"/>, a high-performance time stamp based on <see cref="Stopwatch"/>,
```

Comment:
Yes, it will reflect the changes of `TimeZoneInfo.Local`. The `TimeProvider.System` will always have the exact behavior of system like `DateTimeOffset.Now`.

Reply:
I have added the doc remarks too.